src/text2.c

```diff
@@ -3210,3 +3210,860 @@ void sub_80089AC(const UnkTextStruct2 *r4, UnkTextStruct2_sub *r5_Str)
+"	str r7, [r5, 0x1C]\n"
+"_0800935C:\n"
+"	adds r0, r5, 0\n"
+"	adds r0, 0x20\n"
+"	strb r7, [r0]\n"
+"	b _0800937A\n"
+"	.align 2, 0\n"
+"_08009364: .4byte 0x0000270f\n"
+"_08009368:\n"
+"	cmp r0, 0x7E\n"
+"	bne _0800937A\n"
+"	ldrb r0, [r4, 0x2]\n"
+"	str r0, [r5, 0x2C]\n"
+"	adds r0, r5, 0\n"
+"	adds r0, 0x21\n"
+"	strb r7, [r0]\n"
+"	adds r4, 0x3\n"
+"	b _080091C0\n"
+"_0800937A:\n"
+"	adds r0, r4, 0\n"
+"	pop {r3,r4}\n"
+"	mov r8, r3\n"
+"	mov r9, r4\n"
+"	pop {r4-r7}\n"
+"	pop {r1}\n"
+"	bx r1\n");
+}
+#endif // NONMATCHING
+
+extern s32 gUnknown_202B020;
+extern s32 gUnknown_202B024;
+
+void sub_8009388(void)
+{
+    s32 i, j;
+
+    gUnknown_202B038[0][0][0] = 0xF279;
+    gUnknown_202B038[1][0][0] = 0xF27A;
+    for (i = gUnknown_202B020; i < gUnknown_202B024; i++) {
+        gUnknown_202B038[0][i][0] = 0xF279;
+        gUnknown_202B038[1][i][0] = 0xF27A;
+        for (j = 1; j < 32; j++) {
+            gUnknown_202B038[0][i][j] = 0;
+            gUnknown_202B038[1][i][j] = 0xF27A;
+        }
+    }
+}
+
+void sub_8009408(s32 from, s32 to)
+{
+    s32 i, j;
+
+    gUnknown_202B038[0][0][0] = 0xF279;
+    gUnknown_202B038[1][0][0] = 0xF27A;
+    for (i = from; i < to; i++) {
+        gUnknown_202B038[0][i][0] = 0xF279;
+        gUnknown_202B038[1][i][0] = 0xF27A;
+        for (j = 1; j < 32; j++) {
+            gUnknown_202B038[0][i][j] = 0;
+            gUnknown_202B038[1][i][j] = 0xF27A;
+        }
+    }
+    gUnknown_20274A5 = 1;
+}
+
+UNUSED void sub_8009488(s32 strArrId)
+{
+    s32 i, j;
+    s32 id0, id1;
+    UnkTextStruct1 *strPtr = &gUnknown_2027370[strArrId];
+
+    id0 = strPtr->unk2;
+    for (i = 0; i < strPtr->unk6; i++) {
+        id1 = strPtr->unk0;
+        for (j = 0; j < strPtr->unk4; j++) {
+            gUnknown_202B038[0][id0][id1] &= 0xFC00;
+            gUnknown_202B038[1][id0][id1] = 0xF27A;
+            id1++;
+        }
+        id0++;
+    }
+}
+
+UNUSED void nullsub_172(void) {}
+
+void sub_8009524(s32 strArrId)
+{
+    s32 i, j;
+    UnkTextStruct1 *strPtr = &gUnknown_2027370[strArrId];
+    s32 id0 = strPtr->unk2 - 1;
+
+    for (i = 0; i < strPtr->unk6 + 2; i++) {
+        s32 id1 = strPtr->unk0 - 1;
+        for (j = 0; j < strPtr->unk4 + 2; j++) {
+            gUnknown_202B038[0][id0][id1] &= 0xFC00;
+            gUnknown_202B038[1][id0][id1] = 0xF27A;
+            id1++;
+        }
+        id0++;
+    }
+}
+
+UNUSED void nullsub_173(void) {}
+
+void sub_80095CC(s32 a0, s32 a1)
+{
+    gUnknown_202B020 = a0;
+    gUnknown_202B024 = a1;
+}
+
+UNUSED void nullsub_174(void) {}
+
+s32 sub_80095E4(s32 a0, s32 a1)
+{
+    s32 r1;
+
+    if (a1 == 0)
+        a1 = 10;
+    r1 = a0 * a1;
+
+    if ((r1 % 8) != 0)
+        return (r1 / 8) + 1;
+    else
+        return r1 / 8;
+}
+
+// Same as sub_80095E4 except it doesn't check for a1 == 0
+s32 sub_8009614(s32 a0, s32 a1)
+{
+    s32 r1;
+
+    r1 = a0 * a1;
+
+    if ((r1 % 8) != 0)
+        return (r1 / 8) + 1;
+    else
+        return r1 / 8;
```

Comment:
I was about to comment a way to possibly make this match with neater code, but realized they actually wrote this out like this.
If it was a u32, this would just be a ceiling division. This is not that. It's nonsense for negative numbers.